3rd-party/woocommerce-services.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class WC_Services_Installer {
+
+	public function init() {
+		$this->add_error_notice();
+		$this->try_install();
+	}
+
+	public function try_install() {
+		$action = filter_input( INPUT_GET, 'wc-services-action' );
+
+		if ( 'install' === $action ) {
+			check_admin_referer( 'wc-services-install' );
+
+			$result = $this->install();
+
+			wp_safe_redirect( add_query_arg( array(
+				'wc-services-action'        => false,
```

Comment:
[`menu_page_url()`](https://developer.wordpress.org/reference/functions/menu_page_url/) requires the `$menu_slug` to be specified - the JITM that kicks off this flow will be shown on 3 different pages.

I suppose we could programmatically determine `$menu_slug` and add only the `wc-services-install-error` query arg - just let us know if this is a merge blocker.